lib/api/core/models/security/profile.js

```diff
@@ -10,6 +10,23 @@ function Profile () {
   this.roles = [];
 }
 
+/**
+ * @param {Kuzzle} kuzzle
+ *
+ * @return {Promise}
+ */
+
+Profile.prototype.getRoles = function (kuzzle) {
+  if (!this._roles) {
+    return kuzzle.repositories.role.loadRoles(this.roles)
+      .then(roles => {
+        this._roles = roles;
+        return q(this._roles);
```

Comment:
You don't have to return a promise to make the whole chain resolve `this._roles`. Simply returning the variable will end the chain with this variable as its result.